### PR TITLE
Feature/feature register pet

### PR DESCRIPTION
**Pull request by santidrj**

### CODE REVIEW

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java [hunk 1]
```diff
@@ -12,15 +14,20 @@
...
 
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
 import com.google.android.material.navigation.NavigationView;
 
 import org.pesmypetcare.mypetcare.R;
 import org.pesmypetcare.mypetcare.activities.fragments.NotImplementedFragment;
+import org.pesmypetcare.mypetcare.activities.fragments.RegisterPetCommunication;
+import org.pesmypetcare.mypetcare.activities.fragments.RegisterPetFragment;
 import org.pesmypetcare.mypetcare.databinding.ActivityMainBinding;
+import org.pesmypetcare.mypetcare.features.pets.Gender;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java]: Codacy Issue found: [Unused import - org.pesmypetcare.mypetcare.features.pets.Gender.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5213265)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/RegisterPetFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Pet.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5213265&bid=17094260)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java (hunk 1 above)
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java]: Codacy Issue found: [Avoid unused imports such as 'org.pesmypetcare.mypetcare.features.pets.Gender'](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5213265)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java [hunk 2]
```diff
@@ -12,15 +14,20 @@
...
+import org.pesmypetcare.mypetcare.features.pets.Pet;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java]: Codacy Issue found: [Avoid unused imports such as 'org.pesmypetcare.mypetcare.features.pets.Pet'](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5213265)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java [hunk 3]
```diff
@@ -77,18 +97,33 @@ private void setUpNavigationDrawer() {
...
 
+    /**
+     * Sets up the new fragment.
+     * @param title Title to display in the top bar
+     * @param id Id of the navigation item
+     */
+    private void setUpNewFragment(CharSequence title, int id) {
+        toolbar.setTitle(title);
+
+        if (id == R.id.navigationMyPets)
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java]: Codacy Issue found: ['if' construct must use '{}'s.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5213265)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java [hunk 4]
```diff
@@ -77,18 +97,33 @@ private void setUpNavigationDrawer() {
...
+            floatingActionButton.show();
+        else
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java]: Codacy Issue found: ['else' construct must use '{}'s.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5213265)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java [hunk 5]
```diff
@@ -148,4 +184,25 @@ private void initializeActionDrawerToggle() {
     public boolean onOptionsItemSelected(@NonNull MenuItem item) {
         return actionBarDrawerToggle.onOptionsItemSelected(item) || super.onOptionsItemSelected(item);
     }
+
+    @Override
+    public boolean onKeyDown(int keyCode, KeyEvent event) {
+        if (keyCode == KeyEvent.KEYCODE_BACK) {
+            if (!selectedFragment.equals(APPLICATION_FRAGMENTS[0])) {
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java]: Codacy Issue found: [These nested if statements could be combined](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5213265)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/RegisterPetFragment.java [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+    /**
+     * Sets up the new pet button.
+     */
+    private void setUpAddNewPetButton() {
+        MaterialButton btnNewPet = binding.btnAddPet;
+        btnNewPet.setOnClickListener(v -> {
+            if (isAnyFieldBlank()) {
+                Toast errorMsg = Toast.makeText(getActivity(), R.string.error_empty_field, Toast.LENGTH_LONG);
+                errorMsg.show();
+            }
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/RegisterPetFragment.java]: Codacy Issue found: ['}' at column 13 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5213265)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java (hunk 2 above)
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java]: Codacy Issue found: [Unused import - org.pesmypetcare.mypetcare.features.pets.Pet.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5213265)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Pet.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/RegisterPetFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5213265&bid=17094260)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java [hunk 7]
```diff
@@ -77,18 +95,35 @@ private void setUpNavigationDrawer() {
...
+     * Sets up the new fragment.
+     * @param title Title to display in the top bar
+     * @param id Id of the navigation item
+     */
+    private void setUpNewFragment(CharSequence title, int id) {
+        toolbar.setTitle(title);
+
+        if (id == R.id.navigationMyPets) {
+            floatingActionButton.show();
+        }
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/MainActivity.java]: Codacy Issue found: ['}' at column 9 should be on the same line as the next part of a multi-block statement (one that directly contains multiple blocks: if/else-if/else, do/while or try/catch/finally).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5213265)

**Review by xavier-campos-diaz — APPROVED** [review]

**Review by enrichp — APPROVED** [review]

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/RegisterPetFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Pet.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5213265&bid=17094260)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/RegisterPetFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Pet.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5213265&bid=17094260)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Pet.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/RegisterPetFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5213265&bid=17094260)

**Pull request merged**